clang/lib/AST/DeclBase.cpp

```diff
@@ -1115,7 +1115,9 @@ int64_t Decl::getID() const {
 
 const FunctionType *Decl::getFunctionType(bool BlocksToo) const {
   QualType Ty;
-  if (const auto *D = dyn_cast<ValueDecl>(this))
+  if (const auto *D = dyn_cast<BindingDecl>(this))
```

Comment:
Seems causes the warning. `[-Wunused-but-set-variable]`

Reply:
I'll fix that shortly